lib/spack/spack/build_environment.py

```diff
@@ -817,8 +820,8 @@ def setup_package(pkg, dirty, context='build'):
         # unnecessary
         module('unload', 'cray-libsci')
```

Comment:
This is such a weird line...

Reply:
Way too specific :p

Reply:
Basically all these lines above run on macos / non-cray-linux, where they make no sense whatsoever